amr-wind/equation_systems/tke/source_terms/KransAxell.cpp

```diff
@@ -0,0 +1,126 @@
+    const auto& buoy_prod_arr = (this->m_buoy_prod)(lev).array(mfi);
+    const auto& dissip_arr = (this->m_dissip)(lev).array(mfi);
+    const auto& tke_arr = m_tke(lev).array(mfi);
+    const auto& geom = m_mesh.Geom(lev);
+    const auto& problo = m_mesh.Geom(lev).ProbLoArray();
+    const auto& probhi = m_mesh.Geom(lev).ProbHiArray();
+    const auto& dx = geom.CellSizeArray();
+    const auto& dt = m_time.delta_t();
+    const amrex::Real ref_temp = m_ref_temp;
+    const amrex::Real heat_flux =
+        std::abs(m_gravity[2]) / ref_temp * m_heat_flux;
+    const amrex::Real Cmu = 0.556;
+    const amrex::Real sponge_start = m_sponge_start;
+    const amrex::Real ref_tke = m_ref_tke;
+    const auto tiny = std::numeric_limits<amrex::Real>::epsilon();
+    const amrex::Real kappa = 0.41;
+    const amrex::Real z0 = 0.1;
+    amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+        amrex::Real bcforcing = 0;
+        const amrex::Real ux = vel(i, j, k, 0);
+        const amrex::Real uy = vel(i, j, k, 1);
+        amrex::Real ustar = 0.4;
+        const amrex::Real z = problo[2] + (k + 0.5) * dx[2];
+        if (k == 0) {
+            const amrex::Real m = std::sqrt(ux * ux + uy * uy);
+
+            ustar = m * kappa / std::log(z / z0);
+            const amrex::Real rans_b = std::pow(
+                std::max(heat_flux, 0.0) * kappa * z / std::pow(Cmu, 3),
+                (2.0 / 3.0));
+            bcforcing =
+                (ustar * ustar / (Cmu * Cmu) + rans_b - tke_arr(i, j, k)) / dt;
+        }
+        const amrex::Real zi =
+            std::max((z - sponge_start) / (probhi[2] - sponge_start), 0.0);
+        const amrex::Real sponge_forcing =
+            zi * zi * (tke_arr(i, j, k) - ref_tke);
+        dissip_arr(i, j, k) = std::pow(Cmu, 3) *
+                              std::pow(tke_arr(i, j, k), 1.5) /
+                              (tlscale_arr(i, j, k) + tiny);
+        src_term(i, j, k) += shear_prod_arr(i, j, k) + buoy_prod_arr(i, j, k) -
+                             dissip_arr(i, j, k) - sponge_forcing + bcforcing;
+    });
+    // Add terrain components
+    const bool has_terrain =
+        this->m_sim.repo().int_field_exists("terrain_blank");
+    if (has_terrain) {
+        const auto* const m_terrain_blank =
+            &this->m_sim.repo().get_int_field("terrain_blank");
+        const auto* const m_terrain_drag =
+            &this->m_sim.repo().get_int_field("terrain_drag");
+        const auto& blank_arr = (*m_terrain_blank)(lev).const_array(mfi);
+        const auto& drag_arr = (*m_terrain_drag)(lev).const_array(mfi);
+        amrex::ParallelFor(
+            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                amrex::Real terrainforcing = 0;
+                amrex::Real dragforcing = 0;
+                const amrex::Real ux = vel(i, j, k, 0);
+                const amrex::Real uy = vel(i, j, k, 1);
+                amrex::Real ustar = 0.4;
```

Comment:
Any reason to initialize ustar to be 0.4 here? Looks like you can just `const amrex::Real ustar = m * kappa / std::log(z/z0);` a few lines down

Reply:
Older code not changed.